core/xml.js

```diff
@@ -349,6 +349,13 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       }
     } else if (name == 'shadow') {
       goog.asserts.fail('Shadow block cannot be a top-level block.');
+    }  else if (name == 'variables') {
+      if (i == 1) {
+        Blockly.Xml.domToVariables(xmlChild, workspace);
+      }
+      else {
+        throw Error('\'variables\' tag found in wrong location.');
```

Comment:
Make this more explicit: the variables element must be the first element in the workspace XML.